### PR TITLE
test: "detect" the test number automagically

### PR DESCRIPTION
**Pull request by mrc0mmand**
Specifying the test number manually is tedious and prone to errors (as
recently proven). Since we have all the necessary data to work out the
test number, let's do it automagically.

### CODE REVIEW

**Comment by mrc0mmand** [comment]
This requires https://github.com/systemd/systemd/pull/19383 (since without it TEST-58-REPART is borked).

File: test/units/testsuite-52.sh [hunk 1]
```diff
@@ -2,11 +2,6 @@
 set -ex
 set -o pipefail
 
-if ! test -x /usr/lib/systemd/tests/testdata/units/test-honor-first-shutdown.sh ; then
-        echo "honor-first-shutdown script not found - FAIL" >/testok
```
Comment (keszybz) [inline test/units/testsuite-52.sh:6]: Wat? This this even work?

**Review by keszybz — COMMENTED** [review]
Looks great (and obvious in hindsight ;)). Maybe let's merge #19383 first and retrigger this?

**Comment by mrc0mmand** [comment]
> Maybe let's merge #19383 first and retrigger this?

Definitely, that's what I'm waiting for (since there will be a couple of merge conflicts).

**Review by bluca — APPROVED** [review]
LGTM

**Comment by keszybz** [comment]
bionic-s390x failed in "boot-and-services", should be unrelated.

**Pull request merged**